src/api/devices.ts

```diff
@@ -16,11 +16,22 @@ const sortDevices = (devices: Device[]): Device[] => {
   })
 }
 
+const populateFetchedAt = <T>(item: T): WithFetchedAt<T> => {
+  return {
+    ...item,
+    fetched_at: Math.floor(Date.now() / 1000),
+  }
+}
+
 export const getDevice = async (dongleId: string) =>
   fetcher<Device>(`/v1.1/devices/${dongleId}/`)
+    .then(populateFetchedAt)
```

Comment:
can `fetcher` always do this?

Reply:
yep, nice idea. assuming the api only returns objects and arrays of objects this should work. in the case its a primitive, I throw an error, but lmk and I can change.

Reply:
I had fetcher return `<WithFetchedAt<T>>` instead of the union of `<WithFetchedAt<T>>` and `<WithFetchedAt<T>[]>`, so that we didn't have to perform additional checks throughout the project to determine whether we were dealing with an array or not.